File: 250704/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

let [n, r, c] = input[0].split(' ').map(Number);
let grid = input.slice(1, n + 1).map(line => line.split(' ').map(Number));

// Please Write your code here.
[r, c] = [r - 1, c - 1];

const [dr, dc] = [[-1, 1, 0, 0], [0, 0, -1, 1]];

const answerArr = [grid[r][c]];

function solution() {
    let isMoved = true;

    while (isMoved) {
        isMoved = move();
    }

    answerArr.forEach((elem) => {
        process.stdout.write(`${elem} `);
    })
}

function move() {
    for (let dirNum = 0; dirNum < 4; dirNum++) {
        const [nr, nc] = [r + dr[dirNum], c + dc[dirNum]];

        if (inRange(nr, nc) && grid[nr][nc] > grid[r][c]) {
            [r, c] = [nr, nc];
            answerArr.push(grid[r][c]);
            return true;
        }
    }

    return false;
}

function inRange(r, c) {
    return r >= 0 && r < n && c >= 0 && c < n;
}

solution();

File: 250718/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.py
n, r, c = map(int, input().split())
grid = [[0] * (n) for _ in range(n)]

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

# Please write your code here.
drs, dcs = [-1, 1, 0, 0], [0, 0, 1, -1]
r, c = r - 1, c - 1


def move():
    global r, c

    for i in range(4):
        dr, dc = drs[i], dcs[i]
        nr, nc = r + dr, c + dc

        if (inRange(nr, nc) and grid[r][c] < grid[nr][nc]):
            r, c = nr, nc
            return True
    
    return False



def inRange(r, c):
    return r >= 0 and r < n and c >= 0 and c < n


def solution():
    while True:
        print(grid[r][c], end=' ')

        moved = move()

        if not moved:
            break


# t번만큼 move -> move을 t회 반복
solution()